### PR TITLE
Honor buildid (if specified) for unparsed submission workflow

### PR DESCRIPTION
**Pull request by zackgalbreath**

### CODE REVIEW

**Review by williamjallen — APPROVED** [review]
The general concept here seems reasonable to me once PHPStan is happy.  In the future it would be good to use a relationship between builds and build files, once that relationship exists.

How hard would it be to add a test for this?  This seems like a good edge case to test.

File: app/cdash/tests/CMakeLists.txt [hunk 1]
```diff
@@ -350,8 +350,11 @@ set_tests_properties(uploadfile PROPERTIES DEPENDS projectxmlsequence)
 add_php_test(actualbranchcoverage)
 set_tests_properties(actualbranchcoverage PROPERTIES DEPENDS uploadfile)
 
+add_php_test(unparsedsubmissionshonorbuildid)
+set_tests_properties(unparsedsubmissionshonorbuildid PROPERTIES DEPENDS actualbranchcoverage)
+
 add_php_test(multicoverage)
-set_tests_properties(multicoverage PROPERTIES DEPENDS actualbranchcoverage)
+set_tests_properties(multicoverage PROPERTIES DEPENDS unparsedsubmissionshonorbuildid)
```
Comment (williamjallen) [inline app/cdash/tests/CMakeLists.txt:357]: Can any of these tests be run in parallel?
